airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -279,20 +311,43 @@ else if (file.isAudio()) {
         }
     }
 
-    private void updateAlbum(MediaFile file, MusicFolder musicFolder, Instant lastScanned, Map<String, Integer> albumCount) {
+    private void updateAlbum(MediaFile file, MusicFolder musicFolder, Instant lastScanned, Map<String, AtomicInteger> albumCount, Map<MediaFile, Album> albums) {
         String artist = file.getAlbumArtist() != null ? file.getAlbumArtist() : file.getArtist();
         if (file.getAlbumName() == null || artist == null || file.getParentPath() == null || !file.isAudio()) {
             return;
         }
+        
+        final AtomicBoolean firstEncounter = new AtomicBoolean(false);
+        Album album = albums.compute(file, (k,v) -> {
+            Album a = v;
+            
+            if (a == null) {
+                a = albumDao.getAlbumForFile(k);
+            }
```

Comment:
only fetch once for the first thread to get here.
thereafter all threads will be able to get the in-memory model